code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -214,102 +216,36 @@
 /mob/living/carbon/Xenomorph/handle_regular_hud_updates()
 	if(!client)
 		return FALSE
+
+	// Sanity checks
+	if(!maxHealth)
+		stack_trace("[src] called handle_regular_hud_updates() while having [maxHealth] maxHealth.")
+		return
+	if(!xeno_caste.plasma_max)
```

Comment:
This will only handle 0 and null, will also runtime if xeno_caste is null though that is not an issue as it would be a sign of something much worse happening